fluent-react/README.md

```diff
@@ -0,0 +1,129 @@
+# fluent-react
+
+`fluent-react` provides React bindings for Project Fluent, a localization
+framework designed to unleash the expressive power of the natural language.
+
+
+## Installation
+
+`fluent-react` can be used both on the client-side and the server-side.  You
+can install it from the npm registry or use it as a standalone script (as the
+`FluentReact` global).
+
+    npm install fluent-react
+
+
+## How to use
+
+`fluent-react` takes advantage of React's Components system and the virtual
+DOM.  Translations are stored using the Fluent syntax ([FTL][]) and exposed to
+components via the _provider_ pattern.
+
+An example FTL file might look like the following:
+
+```
+hello-world = Hello, world!
+welcome = Welcome, $username.
+```
+
+
+Wrap your localizable elements in `LocalizedElement`:
+
+```javascript
+import { LocalizedElement } from 'fluent-react';
+
+<LocalizedElement id="hello-world">
+    <p>Hello, world!</p>
+</LocalizedElement>
+```
+
+The `id` prop should be the unique identifier of the translation.  Any
+attributes found in the translation will be applied to the wrapped element.
+You can also pass arguments to the translation as $-prefixed props on
+`LocalizedElement`:
+
+```javascript
+<LocalizedElement id="welcome" $username={name}>
+    <p>{'Welcome, {$username}'}</p>
+</LocalizedElement>
+```
+
+Notice how in the above example the text content of the `<p>` element is
+a string expression, not a text node.  It is provided as help for the developer
+and may be used in the future to extract the source copy. For now, however,
+it's not used for anything.  By putting the text in a simple string expression
+we're saving React's cycles: `<p>Welcome, { $username }</p>` would be rendered
+by React before being passed to `LocalizedElement` where it would be replaced
+by the translation anyways.
```

Comment:
I'm wondering if we have to do that.

It seems to me like that's a minor glitch between Fluent and React and React could solve it by allowing us to take precedence and resolve the localized content before rendering. I'd love to avoid having to do that just because React cannot allow us to execute code in the right order of operations.